---
title: "General-Pipeline"
author: "Ralph Khalife"
date: "2025-11-24"
output: html_document
---

```{r}
# ---- Setup ----
library(TCGAbiolinks)
library(SummarizedExperiment)
library(dplyr)
library(edgeR)
library(ggplot2)
library(survival)
library(survminer)

dir.create("figs", showWarnings = FALSE)
dir.create("results", showWarnings = FALSE)

# ---- Configure Cancer Type Here ----
project_id <- "TCGA-SKCM"    # <--- CHANGE THIS EACH TIME

# ---- Gene ID (EZR) ----
ezr_id <- "ENSG00000092820"

```

```{r}
# ---- Step 1: Download data for one TCGA project ----

query <- GDCquery(
  project = project_id,
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "STAR - Counts"
)

GDCdownload(query)
se <- GDCprepare(query)
```

```{r}
# ---- Step 2: Extract EZR expression + normalize ----
expr  <- assay(se)
meta  <- as.data.frame(colData(se))

# robust match for EZR row (handles version suffix)
ezr_row <- grep(paste0("^", ezr_id), rownames(expr), value = TRUE)[1]

EZR_counts <- expr[ezr_row, ]

# Normalization
dge <- DGEList(counts = expr)
dge <- calcNormFactors(dge)
logcpm <- cpm(dge, log = TRUE, prior.count = 1)

EZR_logcpm <- logcpm[ezr_row, ]

meta$barcode_short <- substr(meta$barcode, 1, 12)

df_ezr <- data.frame(
  barcode    = meta$barcode_short,
  SampleType = meta$sample_type,
  RawCount   = as.numeric(EZR_counts),
  logCPM     = as.numeric(EZR_logcpm)
)

head(df_ezr)
```
```{r}
# ---- Step 3: Expression: tumor vs normal ----
p_box <- ggplot(df_ezr, aes(x = SampleType, y = logCPM, fill = SampleType)) +
  geom_boxplot() +
  theme_bw() +
  labs(
    title = paste("EZR Expression in", project_id, "(log2-CPM)"),
    x = "",
    y = "log2(CPM)"
  ) +
  theme(legend.position = "none")

p_box

ggsave(paste0("figs/EZR_", project_id, "_Boxplot.png"),
       p_box, width = 6, height = 4)

summary_ezr <- df_ezr %>%
  group_by(SampleType) %>%
  summarise(
    n = n(),
    mean_logCPM = mean(logCPM, na.rm = TRUE),
    median_logCPM = median(logCPM, na.rm = TRUE),
    sd_logCPM = sd(logCPM, na.rm = TRUE)
  )

write.csv(summary_ezr,
          paste0("results/EZR_", project_id, "_ExpressionSummary.csv"),
          row.names = FALSE)

```

```{r}
# ---- Step 4: Clinical data + merge ----

clinical <- GDCquery_clinic(project_id, type = "clinical")

clinical_clean <- clinical %>%
  select(bcr_patient_barcode, vital_status,
         days_to_death, days_to_last_follow_up) %>%
  mutate(
    time = ifelse(!is.na(days_to_death), days_to_death, days_to_last_follow_up),
    status = ifelse(vital_status == "Dead", 1, 0)
  ) %>%
  filter(!is.na(time))

surv_df <- merge(
  df_ezr[, c("barcode", "logCPM")],
  clinical_clean,
  by.x = "barcode", by.y = "bcr_patient_barcode"
)

surv_df <- surv_df[!duplicated(surv_df$barcode), ]

# median cutoff
cutoff <- median(surv_df$logCPM, na.rm = TRUE)
surv_df$Group <- ifelse(surv_df$logCPM > cutoff, "High", "Low")

fit <- survfit(Surv(time, status) ~ Group, data = surv_df)

km_plot <- ggsurvplot(
  fit, data = surv_df,
  pval = TRUE, conf.int = TRUE,
  risk.table = TRUE, risk.table.col = "strata",
  palette = c("#E7B800", "#2E9FDF"),
  title = paste("KM Survival Curve for EZR in", project_id),
  xlab = "Days", ylab = "Overall Survival Probability",
  surv.median.line = "hv"
)

ggsave(paste0("figs/EZR_", project_id, "_KMcurve.png"),
       km_plot$plot, width = 6, height = 5)

```


```{r}
# ---- Step 5: Cox regression ----

cox <- coxph(Surv(time, status) ~ logCPM, data = surv_df)
cox_sum <- summary(cox)

cox_output <- data.frame(
  Cancer = project_id,
  Gene = "EZR",
  HR = cox_sum$coefficients[, "exp(coef)"],
  HR_lower95 = cox_sum$conf.int[, "lower .95"],
  HR_upper95 = cox_sum$conf.int[, "upper .95"],
  p_value = cox_sum$coefficients[, "Pr(>|z|)"],
  z_score = cox_sum$coefficients[, "z"]
)

write.csv(cox_output,
          paste0("results/EZR_", project_id, "_Cox.csv"),
          row.names = FALSE)

cox_output

```



