---
title: "Bulk Run"
author: "Ralph Khalife"
date: "2025-11-24"
output: html_document
---

```{r}
library(TCGAbiolinks)
library(SummarizedExperiment)
library(edgeR)
library(survival)
library(survminer)
library(dplyr)
library(ggplot2)

# PODXL ENSG ID
podxl_id <- "ENSG00000128567"

# TCGA cancers to analyze
cancers <- c(
  "TCGA-BRCA", "TCGA-LUAD", "TCGA-LUSC",
  "TCGA-COAD", "TCGA-READ", "TCGA-STAD", "TCGA-GBM",
  "TCGA-OV",   "TCGA-HNSC", "TCGA-PAAD"
)

dir.create("bulk_results", showWarnings = FALSE)

run_podxl_analysis <- function(project){

  message("\n=======================")
  message("Analyzing ", project)
  message("=======================\n")

  # ---- 1. Download data ----
  query <- GDCquery(
    project = project,
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification",
    workflow.type = "STAR - Counts"
  )

  GDCdownload(query)
  se <- GDCprepare(query)

  expr <- assay(se)
  genes <- rownames(expr)
  meta <- as.data.frame(colData(se))
  meta$barcode <- substr(meta$barcode, 1, 12)

  if(!(podxl_id %in% genes)){
    message("PODXL not found in ", project)
    return(NULL)
  }

  # ---- 2. log2-CPM normalization ----
  dge <- DGEList(counts = expr)
  dge <- calcNormFactors(dge)
  logcpm <- cpm(dge, log = TRUE, prior.count = 1)

  podxl_logcpm <- logcpm[podxl_id, ]
  df <- data.frame(
    barcode = substr(colnames(expr), 1, 12),
    SampleType = meta$sample_type,
    Expression = podxl_logcpm
  )

  # ---- 3. BOX PLOT ----
  p_box <- ggplot(df, aes(x = SampleType, y = Expression, fill = SampleType)) +
    geom_boxplot() + theme_bw() +
    labs(title = paste("PODXL Expression in", project),
         y = "log2(CPM)")

  ggsave(paste0("bulk_results/", project, "_PODXL_Boxplot.png"),
         p_box, width = 6, height = 4)

  # ---- 4. SURVIVAL ----
  clinical <- GDCquery_clinic(project, "clinical")
  clinical2 <- clinical %>%
    select(bcr_patient_barcode, vital_status,
           days_to_death, days_to_last_follow_up) %>%
    mutate(
      time = ifelse(!is.na(days_to_death), days_to_death, days_to_last_follow_up),
      status = ifelse(vital_status == "Dead", 1, 0)
    ) %>% filter(!is.na(time))

  surv_df <- merge(df, clinical2,
                   by.x = "barcode", by.y = "bcr_patient_barcode")
  surv_df <- surv_df[!duplicated(surv_df$barcode), ]

  if(nrow(surv_df) < 30){
    message("Not enough survival data for ", project)
    return(NULL)
  }

  cutoff <- median(surv_df$Expression, na.rm=TRUE)
  surv_df$Group <- ifelse(surv_df$Expression > cutoff, "High", "Low")

  fit <- survfit(Surv(time, status) ~ Group, data = surv_df)

  p_km <- ggsurvplot(
    fit, data = surv_df, pval = TRUE,
    conf.int = TRUE, risk.table = TRUE,
    palette = c("#E7B800", "#2E9FDF"),
    title = paste("KM for PODXL in", project)
  )

  ggsave(paste0("bulk_results/", project, "_PODXL_KM.png"),
         p_km$plot, width = 6, height = 5)

  # ---- 5. COX MODEL ----
  cox <- coxph(Surv(time, status) ~ Expression, data = surv_df)
  hr <- summary(cox)$coefficients

  write.csv(hr,
            paste0("bulk_results/", project, "_PODXL_Cox.csv"),
            row.names = FALSE)

  message("Finished ", project)
}

# ---- Run on all cancers ----
lapply(cancers, run_podxl_analysis)

```

